docs/errors_overview.md

```diff
@@ -0,0 +1,53 @@
+# XLA Errors Overview
+
+XLA errors are categorized into different XLA error sources. Each source has
+a list of an additional context other than the error message, which will be
+attached to each error within the category.
+
+### E01xx - Runtime data transfer/allocation failures
```

Comment:
We might be restricting ourselves by telling people how we plan to group these errors, maybe skip these sections completely for now?